eng/Versions.props

```diff
@@ -30,7 +30,7 @@
     <RoslynDiagnosticsNugetPackageVersion>3.3.3-beta1.21105.3</RoslynDiagnosticsNugetPackageVersion>
     <MicrosoftCodeAnalysisNetAnalyzersVersion>6.0.0-rc1.21366.2</MicrosoftCodeAnalysisNetAnalyzersVersion>
     <MicrosoftCodeAnalysisTestingVersion>1.1.0-beta1.21322.2</MicrosoftCodeAnalysisTestingVersion>
-    <CodeStyleAnalyzerVersion>3.10.0</CodeStyleAnalyzerVersion>
+    <CodeStyleAnalyzerVersion>4.0.0-3.final</CodeStyleAnalyzerVersion>
```

Comment:
💡 The version of dotnet-format in dotnet-tools.json should be updated to match this version at the same time. @JoeRobich can help identify the matching version number.

Reply:
Done. Thanks